vault/request_handling.go

```diff
@@ -641,9 +642,16 @@ func (c *Core) handleRequest(ctx context.Context, req *logical.Request) (retResp
 	resp, routeErr := c.router.Route(ctx, req)
 	// If we're replicating and we get a read-only error from a backend, need to forward to primary
 	if routeErr != nil {
+		if shouldForward(c, routeErr) {
```

Comment:
Should this logic be inverted? I thought we'd only count the requests that we're not forwarding?

Reply:
Whoops, good catch.